Source/Core/DolphinQt2/GameList/GameFile.cpp

```diff
@@ -291,6 +296,53 @@ QString GameFile::GetLanguage(DiscIO::Language lang) const
   }
 }
 
+bool GameFile::IsInstalled() const
+{
+  _assert_(m_platform == DiscIO::Platform::WII_WAD);
+
+  const std::string content_dir =
+      Common::GetTitleContentPath(m_title_id, Common::FromWhichRoot::FROM_CONFIGURED_ROOT);
+
+  if (!File::IsDirectory(content_dir))
+    return false;
+
+  // Since this isn't IOS and we only need a simple way to figure out if a title is installed,
+  // we make the (reasonable) assumption that having more than just the TMD in the content
+  // directory means that the title is installed.
+  const auto entries = File::ScanDirectoryTree(content_dir, false);
+  return std::any_of(entries.children.begin(), entries.children.end(),
+                     [](const auto& file) { return file.virtualName != "title.tmd"; });
+}
+
+bool GameFile::Install()
+{
+  _assert_(m_platform == DiscIO::Platform::WII_WAD);
+
+  return DiscIO::CNANDContentManager::Access().Install_WiiWAD(m_path.toStdString());
+}
+
+bool GameFile::Uninstall()
+{
+  _assert_(m_platform == DiscIO::Platform::WII_WAD)
```

Comment:
It compiles because it expands to:

```
if (!(m_platform == DiscIO::Platform::WII_WAD))
{
  if (!PanicYesNo("An error occurred.\n\n  Line: %d\n  File: %s\n\nIgnore and continue?", __LINE__, __FILE__))
    Crash();
}
```